src/ast/passes/semantic_analyser.cpp

```diff
@@ -37,9 +37,57 @@ static const std::map<std::string, std::tuple<size_t, bool>> &getIntcasts()
   return intcasts;
 }
 
+static std::pair<uint64_t, uint64_t> getUIntTypeRange(const SizedType &ty)
+{
+  assert(ty.IsIntegerTy());
+  auto size = ty.GetSize();
+  if (size == 1) {
```

Comment:
Nit: it feels like a `switch` statement would make the code more readable here but that may be just my preference. Same in the other function below.